### PR TITLE
Simpler output

### LINKED ISSUE #748: Improve reek's default messages
I believe our default messages are bad since they are catered to people who kind of know what to do anyway.
I believe right now we are deterring reek beginners - we should make `Reek` so beginner-friendly that my grandmother would fully understand what our error messages mean without special switch by default. OR, and this is what I'm actually suggesting, link to our helpful doc pages right away because I believe that if you have a concise message with a link at the end, most people will click this.

E.g. I'd change this: 

>>   [5, 5]:ShoppingCart#gross_price refers to item more than self (maybe move it to another class?) (FeatureEnvy)

to:

>> [Line 5]: ShoppingCart#gross_price smells of [FeatureEnvy](http://bit.ly/1QSFryc)

I talked to quite some `Reek` users recently and believe it or not, almost NOBODY had even realized that we had very helpful doc pages which not only explain the problem, but also often suggest a solution.

WDYT?

### PR DESCRIPTION
**Pull request by chastell**
This is an attempt to fix #748 as per discussion therein.

### CODE REVIEW

File: features/samples.feature [hunk 1]
```diff
@@ -12,272 +12,274 @@ Feature: Basic smell detection
...
+      ControlParameter: Inline::C#parse_signature is controlled by argument raw
+      DataClump: Inline::C takes parameters [options, src] to 5 methods
+      DuplicateMethodCall: Inline#self.rootdir calls env.nil? 2 times
+      DuplicateMethodCall: Inline::C#build calls $? != 0 2 times
+      DuplicateMethodCall: Inline::C#build calls Inline.directory 5 times
+      DuplicateMethodCall: Inline::C#build calls io.puts 6 times
+      DuplicateMethodCall: Inline::C#build calls io.puts("#endif") 2 times
+      DuplicateMethodCall: Inline::C#build calls io.puts("#ifdef __cplusplus") 2 times
+      DuplicateMethodCall: Inline::C#build calls warn("Output:
+    #{result}") 2 times
```
Comment (mvz) [inline features/samples.feature]: Let's use `\n` here like before.

File: features/samples.feature [hunk 2]
```diff
@@ -12,272 +12,274 @@ Feature: Basic smell detection
...
+      DuplicateMethodCall: Inline::C#crap_for_windoze calls Config::CONFIG["libdir"] 2 times
+      DuplicateMethodCall: Inline::C#generate calls result.sub!(/\A
+    /, "") 2 times
```
Comment (mvz) [inline features/samples.feature]: Let's use `\n` here like before.

**Comment by mvz** [comment]
Looks good. Am I right that this only goes part of the way toward getting #748 done?

**Comment by troessner** [comment]
Looks great to me! But what about the helpful link at the end of each message?
I thought putting this link there would be default?

**Comment by chastell** [comment]
> But what about the helpful link at the end of each message?
> I thought putting this link there would be default?

Oh, I must’ve misunderstood [your exchange from #748](https://github.com/troessner/reek/issues/748#issuecomment-154065621):

>> I'd like to keep `-U` semantics as it is for now and revisit that part later.

> Ok, agreed.

So, just to be clear: we want to keep `-U` in Reek 3 but switch the default to `true`, right?

**Comment by troessner** [comment]
Yes, let's make it default!
My working hypothesis: 99% of all Reek beginners will not be like "ok, no idea what the hell FeatureEnvy is but hey, LETS CHECK IF THERE IS A FLAG THAT EXPLAINS IT!" but rather like "ok, I'm outta here, f**k that". 
I'm pretty sure nobody knows about this flag or is using it.

We gotta make it easy for most people snd not for the few ones that know Reek by heart.

**Comment by mvz** [comment]
Now whatever happened to that feature for being able to specify flags in the configuration file?

**Comment by mvz** [comment]
> Yes, let's make it default!

I see your point. Let's do this!

**Comment by chastell** [comment]
Roger! I’m on it. Should we keep the square brackets around the URL?

```
[1]:IrresponsibleModule: Dirty has no descriptive comment [https://github.com/troessner/reek/blob/master/docs/Irresponsible-Module.md]
[2]:UncommunicativeMethodName: Dirty#a has the name 'a' [https://github.com/troessner/reek/blob/master/docs/Uncommunicative-Method-Name.md]
[3]:UncommunicativeMethodName: Dirty#b has the name 'b' [https://github.com/troessner/reek/blob/master/docs/Uncommunicative-Method-Name.md]
```

vs

```
[1]:IrresponsibleModule: Dirty has no descriptive comment https://github.com/troessner/reek/blob/master/docs/Irresponsible-Module.md
[2]:UncommunicativeMethodName: Dirty#a has the name 'a' https://github.com/troessner/reek/blob/master/docs/Uncommunicative-Method-Name.md
[3]:UncommunicativeMethodName: Dirty#b has the name 'b' https://github.com/troessner/reek/blob/master/docs/Uncommunicative-Method-Name.md
```

**Comment by troessner** [comment]
:+1: on the square brackets!

**Comment by mvz** [comment]
:+1: on the square brackets as well!

Also, can we get rid of the first colon, so we only have a colon after the smell type, not before? Like so:

```
[1] IrresponsibleModule: Dirty has no descriptive comment [https://github.com/troessner/reek/blob/master/docs/Irresponsible-Module.md]
[2] UncommunicativeMethodName: Dirty#a has the name 'a' [https://github.com/troessner/reek/blob/master/docs/Uncommunicative-Method-Name.md]
[3] UncommunicativeMethodName: Dirty#b has the name 'b' [https://github.com/troessner/reek/blob/master/docs/Uncommunicative-Method-Name.md]
```

**Comment by chastell** [comment]
> Also, can we get rid of the first colon, so we only have a colon after the smell type, not before?

Hm, we [discussed it](https://github.com/troessner/reek/issues/748#issuecomment-150773746) and decided to keep the `[…comma-separated line numbers…]:…text with a CamelCased smell name…` in Reek 3 for the sake of potential shell scripts that parse this and go with `Line X: …text…` / `Lines X, Y: …text…` in Reek 4.

That said, if you both are ok with changes to this in Reek 3 I’m perfectly fine with making them – just let me know so I won’t have to adjust all of our hard-coded Cucumber examples for the third time, as it’s rather daunting. ;)

**Comment by mvz** [comment]
Hm, ok, let's leave it like this and not make things more complicated for ourselves right now.

**Comment by troessner** [comment]
>> Hm, ok, let's leave it like this and not make things more complicated for ourselves right now.

:+1:

**Comment by troessner** [comment]
>> That said, if you both are ok with changes to this in Reek 3 I’m perfectly fine with making them – just let me know so I won’t have to adjust all of our hard-coded Cucumber examples for the third time, as it’s rather daunting. ;)

I honestly thought you enjoyed things like that :laughing:

**Comment by chastell** [comment]
Done!

> I honestly thought you enjoyed things like that :laughing: 

Oh I do enjoy stuff like _fix all the ivar access places so my eyes won’t bleed on a daily basis_; stuff like _let’s go and fix all of our way-over-specified, slow-as-molasses Cucumber scenarios_… not as much. :wink: 

(Protip for the future: running `cucumber --fail-fast --order random` makes such tasks so much more exciting! `\o/`)

**Comment by troessner** [comment]
@chastell I'm a little confused now, is this PR ready for final review?

**Comment by chastell** [comment]
Yes, yes it is!

**Comment by troessner** [comment]
Looks awesome! Can you squash pls? Then I'd happily merge and then we should get a new Reek version out there asap ;)

**Comment by JuanitoFatas** [comment]
@troessner If this is ok to merge, why not do a

`git am -3 https://github.com/troessner/reek/pull/765`

against latest master :smile: ?

**Comment by troessner** [comment]
:D

**Comment by chastell** [comment]
> Can you squash pls?

Hm, there are three distinct commits doing three distinct things (one does no-change clean-ups, one reorders the messages, one switches `--wiki-links`’ default). Do you really want to squash them into one?

(I rebased this against master and made sure the Code Climate example is adjusted for these changes.) :)

**Comment by troessner** [comment]
>> Hm, there are three distinct commits doing three distinct things (one does no-change clean-ups, one reorders the messages, one switches --wiki-links’ default). Do you really want to squash them into one?

Merged!

Care for a PR with the version bump? ;)

**Pull request merged**